javalib/src/main/scala/java/lang/Double.scala

```diff
@@ -244,7 +244,7 @@ object Double {
       errno.errno = 0
       val res = stdlib.strtod(cstr, end)
 
-      if (errno.errno == 0) res
+      if (errno.errno == 0 && cstr != !end) res
```

Comment:
This is much better than what we had before, and it indeed fixes `parseDouble("foobar")` to throw as expected. But the problem still remains for the `parseFloat("0.0foobar")`. `strtod` would happily eat `0.0` and then stop.

We can fix it by: 
1. Finding pointer to the end of the string by skipping all characters until we see `'\0'`. 
2. And then checking that `end` as returned by `strtod` is the end of the string.

Reply:
Thanks - I'll get this fixed. Yeah, I think that'll work @ekrich - thanks for the suggestion.